x-pack/solutions/observability/plugins/slo/server/services/tasks/temp_summary_cleanup_task.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { errors } from '@elastic/elasticsearch';
+import { type CoreSetup, type Logger, type LoggerFactory } from '@kbn/core/server';
+import { ConcreteTaskInstance, TaskManagerSetupContract } from '@kbn/task-manager-plugin/server';
+import { getDeleteTaskRunResult } from '@kbn/task-manager-plugin/server/task';
+import { SLOConfig, SLOPluginStartDependencies } from '../../types';
+import { CleanUpTempSummary } from '../management/clean_up_temp_summary';
+
+export const TYPE = 'slo:temp-summary-cleanup-task';
+export const VERSION = '1.0.0';
+
+interface TaskSetupContract {
+  taskManager: TaskManagerSetupContract;
+  core: CoreSetup;
+  logFactory: LoggerFactory;
+  config: SLOConfig;
+}
+
+export class TempSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private config: SLOConfig;
+  private wasStarted: boolean = false;
+
+  constructor(setupContract: TaskSetupContract) {
+    const { core, config, taskManager, logFactory } = setupContract;
+    this.logger = logFactory.get(this.taskId);
+    this.config = config;
+
+    this.logger.info('Registering task with [2m] timeout');
+
+    taskManager.registerTaskDefinitions({
+      [TYPE]: {
+        title: 'SLO temp summary cleanup task',
+        timeout: '2m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask(taskInstance, core);
+            },
+
+            cancel: async () => {
+              this.abortController.abort('Timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  public async start(plugins: SLOPluginStartDependencies) {
+    const hasCorrectLicense = (await plugins.licensing.getLicense()).hasAtLeast('platinum');
+    if (!hasCorrectLicense) {
+      this.logger.debug('Platinum license is required');
```

Comment:
```suggestion
      this.logger.debug('Platinum license is required to enable TempSummaryCleanupTask ');
```

Reply:
The logger is configured to prefixed the log message with the name of the task